core/model/src/main/java/org/eclipse/rdf4j/model/util/ModelBuilder.java

```diff
@@ -0,0 +1,360 @@
+	}
+
+	/**
+	 * Add an RDF statement with the predicate and object to the model, using the current subject and graph
+	 * (either named or default).
+	 * 
+	 * @param predicate
+	 *        the statement's predicate.
+	 * @param object
+	 *        the statement's object. If the supplied object is a {@link BNode}, {@link IRI}, or
+	 *        {@link Literal}, the object is used directly. Otherwise, this method creates a typed
+	 *        {@link Literal} out of the supplied object, mapping the runtime type of the object to the
+	 *        appropriate XML Schema type. If no mapping is available, the method creates a literal with the
+	 *        string representation of the supplied object as the value, and {@link XMLSchema#STRING} as the
+	 *        datatype. Recognized types are {@link Boolean} , {@link Byte}, {@link Double}, {@link Float},
+	 *        {@link Integer}, {@link Long}, {@link Short}, {@link XMLGregorianCalendar } , and {@link Date}.
+	 * @return this {@link ModelBuilder}
+	 * @throws ModelException
+	 *         if the current subject is not set using {@link #subject(Resource)} or {@link #subject(String)}.
+	 */
+	public ModelBuilder add(IRI predicate, Object object) {
+		if (currentSubject == null) {
+			throw new ModelException("subject not set");
+		}
+		return add(currentSubject, predicate, object);
+	}
+
+	/**
+	 * Add an RDF statement with the predicate and object to the model, using the current subject and graph
+	 * (either named or default).
+	 * 
+	 * @param predicate
+	 *        the statement's predicate. This can be defined either as a prefixed name string (e.g.
+	 *        "ex:john"), or as a full IRI (e.g. "http://example.org/john"). If supplied as a prefixed name,
+	 *        the {@link ModelBuilder} will need to have a namespace mapping for the prefix.
+	 * @param object
+	 *        the statement's object. If the supplied object is a {@link BNode}, {@link IRI}, or
+	 *        {@link Literal}, the object is used directly. Otherwise this method creates a typed
+	 *        {@link Literal} out of the supplied object, mapping the runtime type of the object to the
+	 *        appropriate XML Schema type. If no mapping is available, the method creates a literal with the
+	 *        string representation of the supplied object as the value, and {@link XMLSchema#STRING} as the
+	 *        datatype. Recognized types are {@link Boolean} , {@link Byte}, {@link Double}, {@link Float},
+	 *        {@link Integer}, {@link Long}, {@link Short}, {@link XMLGregorianCalendar } , and {@link Date}.
+	 * @return this {@link ModelBuilder}
+	 * @throws ModelException
+	 *         if the current subject is not set using {@link #subject(Resource)} or {@link #subject(String)}.
+	 */
+	public ModelBuilder add(String predicate, Object object) {
+		return add(mapToIRI(predicate), object);
+	}
+
+	/**
+	 * Return the created {@link Model}
+	 * 
+	 * @return the {@link Model}
+	 */
+	public Model build() {
+		return model;
```

Comment:
I was okay with that when I reviewed it, based on the Javadoc for the constructor where they would submit the model. If you wanted another option, a "buildImmutable()" or similar version that returns an unmodifiable copy of the Model may be useful, but the usual model for Java collections is to be mutable by default. 

In addition, if they submit a Model, we have already been modifying it before this point, so we would need to take the copy in the constructor rather than here.